client/api_client.go

```diff
@@ -79,28 +77,16 @@ type Option struct {
 // The grpcAddress indicate the grpc address of greenfield chain.
 // The account indicate the account used for signing transactions.
 // The rpcAddress indicate the rpc address of the tendermint client.
-func New(chainID string, grpcAddress string, option Option) (Client, error) {
-	if grpcAddress == "" || chainID == "" {
+func New(chainID string, rpcAddress string, option Option) (Client, error) {
```

Comment:
If we use HTTP protocol, it is recommended to use `endpoint` directly, not rpcAddress.